### ISSUE TITLE
SimProcedure 'strncmp' (strings not equal case)

### ISSUE THREAD

**Issue by frediramos** [angr/angr#2980]
**Describe the 'bug'**
The Sim procedure for the libc function 'strncmp' correctly models all the execution paths that return the value zero, accounting for the cases where both strings are equal ('match_constraint)', both strings are empty ('empty_case'), or the limit argument 'n' is equal to zero ('l0_case').

However, regarding the execution paths that correspond to a return value different that zero, the cases where the input strings differ ('nomatch_case'), this procedure generates the following formula:

```
#Line 154:
nomatch_case = self.state.solver.And(limit != 0, nomatch_constraint, ret_expr == 1)
```

According to strncmp’s specification, this function should also return a negative value when the first string is lower than the second.
By having a fixed return value of 1 for all cases where symbolic strings are different, the procedure leads to both missing and incorrect executions paths. 
Even though the strcmp/strncmp functions are generally only used to check if two strings match, it is still important to take into account that these functions can also be used to determine which string is greater/lower.
For instance, this use case can be found in one of the Challenge Binaries from CGC, in particular, the Challenge: KPRCA_00087 (XStore); XStore/src/tr.c:57

We are aware that Sim procedures should be kept simple, but we believe that every procedure should either contain all correct paths, or not contain any incorrect path, and we have a good idea of how to patch the summary to make either of those conditions hold or even both of them, while maintaining efficiency.

**Comment by rhelmot** [comment]
I totally agree that our simproc implementations should be sound and that this is a big blind spot. I don't particularly have the bandwidth to look into solving this right now, but there's a good chance there is an easy fix in copying code from our memcmp simprocedure, which does seem to implement the trivalued return code correctly.
